### ISSUE TITLE
Configurable Transport

### ISSUE THREAD

**Issue by tarndt** [Azure/azure-storage-file-go#17]
I would like to know if there is a way to configure this library to use a specific [`http.RoundTripper`](https://golang.org/pkg/net/http/#RoundTripper) for its external/HTTP communication? We have a multi-tenant application and each tenant has a specific [`http.RoundTripper`](https://golang.org/pkg/net/http/#RoundTripper) we want requests on the tenants behalf to use.

**Comment by JeffreyRichter** [comment]
Yes, this is possible. Our AzCopy utility does this. See https://github.com/Azure/azure-storage-azcopy/blob/f1a1ee3e99053cd0a0a19c471891ffaa6009d9b8/ste/mgr-JobPartMgr.go#L67. There is a plan to make it a bit simpler; see https://github.com/Azure/azure-storage-blob-go/issues/89.

**Comment by tarndt** [comment]
Thanks! Should I leave this open to track making this easier or would you like me to close?

**Comment by jiacfan** [comment]
Thanks for the suggestion! And it's nice to have it tracking the enhancement, we'll optimize this with a release.
